hal/network/lwip/wiznet/wiznetif_config.cpp

```diff
@@ -0,0 +1,206 @@
+        // if (configData->version == WIZNETIF_CONFIG_DATA_VERSION_V1) {
+        //     // check size, and initialize / move elements as appropriate
+        // }
+        initializeWizNetifConfigData(wizNetifConfigData_);
+        saveWizNetifConfigData();
+    }
+    initialized_ = true;
+}
+
+int WizNetifConfig::initializeWizNetifConfigData(WizNetifConfigData& configData) {
+    memset(&configData, 0, sizeof(wizNetifConfigData_));
+    configData.size = sizeof(WizNetifConfigData);
+    configData.version = WIZNETIF_CONFIG_DATA_VERSION;
+    configData.cs_pin = PIN_INVALID;
+    configData.reset_pin = PIN_INVALID;
+    configData.int_pin = PIN_INVALID;
+
+    return SYSTEM_ERROR_NONE;
+}
+
+void WizNetifConfig::logWizNetifConfigData(WizNetifConfigData& configData) {
+    WZCFG_LOG_DEBUG(INFO, "wizNetifConfigData size:%u ver:%u cs_pin:%u reset_pin:%u int_pin:%u",
+            configData.size,
+            configData.version,
+            configData.cs_pin,
+            configData.reset_pin,
+            configData.int_pin);
+}
+
+int WizNetifConfig::saveWizNetifConfigData() {
+    WizNetifConfigData tempData = {};
+    tempData.size = sizeof(WizNetifConfigData);
+    tempData.version = WIZNETIF_CONFIG_DATA_VERSION;
+    int result = SystemCache::instance().get(SystemCacheKey::WIZNET_CONFIG_DATA, (uint8_t*)&tempData, sizeof(tempData));
+    if (result != std::min(tempData.size, wizNetifConfigData_.size) ||
+            memcmp(&tempData, &wizNetifConfigData_, std::min(tempData.size, wizNetifConfigData_.size)) != 0) {
+        WZCFG_LOG_DEBUG(INFO, "Saving cached wizNetifConfigData, size: %d", wizNetifConfigData_.size);
+        logWizNetifConfigData(wizNetifConfigData_);
+        result = SystemCache::instance().set(SystemCacheKey::WIZNET_CONFIG_DATA, (uint8_t*)&wizNetifConfigData_, sizeof(wizNetifConfigData_));
+    }
+    return (result < 0) ? result : SYSTEM_ERROR_NONE;
+}
+
+int WizNetifConfig::recallWizNetifConfigData() {
+    WizNetifConfigData tempData = {};
+    tempData.size = sizeof(WizNetifConfigData);
+    tempData.version = WIZNETIF_CONFIG_DATA_VERSION;
+    int result = SystemCache::instance().get(SystemCacheKey::WIZNET_CONFIG_DATA, (uint8_t*)&tempData, sizeof(tempData));
+    if (result != std::min(tempData.size, wizNetifConfigData_.size)) {
+        return SYSTEM_ERROR_NOT_FOUND;
+    }
+    WZCFG_LOG_DEBUG(INFO, "Recalling cached wizNetifConfigData, size: %d", result);
+    logWizNetifConfigData(tempData);
+    memcpy(&wizNetifConfigData_, &tempData, sizeof(tempData));
+    return SYSTEM_ERROR_NONE;
+}
+
+// int WizNetifConfig::deleteWizNetifConfigData() {
+//     WZCFG_LOG_DEBUG(INFO, "Deleting cached wizNetifConfigData");
+//     int result = SystemCache::instance().del(SystemCacheKey::WIZNET_CONFIG_DATA);
+//     return (result < 0) ? result : SYSTEM_ERROR_NONE;
+// }
+
+int WizNetifConfig::setConfigData(const WizNetifConfigData* configData) {
+    WizNetifConfigLock lk;
+
+    if (!initialized_) {
```

Comment:
Nit: might want to move the check inside `init()` itself perhaps and just call it?